### ISSUE TITLE
What's a good setup for ambassador to host both private and public mappings?

### ISSUE THREAD

**Issue by ryanoolala** [datawire/ambassador#2258]
## Purpose ##
I am setting up my cluster with ambassador to have a way which
- internet users can access it through `public.my-domain.com`
- vpn users can access it through `private.my-domain.com`

## Issue ##
I have both a public and private load balancer to which i point the domains to respectively. As ambassador on AWS uses classic LB, i am not able to block access using the loadbalancer by checking `Host` headers. Also, my understanding is that if i try to `curl -H "Host: private.my-domain.com" https://public-loadbalancer` with only 1 ambassador deployment serving both loadbalancers i will be able to reach `private.my-domain.com` 

I was wondering if there is a recommended setup to achieve this?

## My plan now ##
- run two deployments of ambassador
- use the `AMBASSADOR_ID` to control the available mappings to the respective pods.
- point the public load balancer service to "public" ambassador pods
- point private load balancer service to "private" ambassador pods

Example mapping:
```yaml
apiVersion: getambassador.io/v2
kind:  Mapping
metadata:
  name:  mapping-public
spec:
  ambassador_id: ambassador-public
  prefix: /
  host: public.my-domain.com
  service: public-service
---
apiVersion: getambassador.io/v2
kind:  Mapping
metadata:
  name:  mapping-private
spec:
  ambassador_id: ambassador-private
  prefix: /
  host: private.my-domain.com
  service: private-service
```

This way the mapping for `private.my-domain.com` won't be valid for routes going through `public.my-domain.com` -> "Public" load balancer -> "Public" ambassador pods.

Are there other solutions available for this? maybe through some Mapping configuration feature that is available but i'm unaware of.

Thank you for your time :)

**Comment by ryanoolala** [comment]
No comments? :(

**Comment by adriangudas** [comment]
New user here, but I had a similar situation, and I ended up using separate ports (using TCPMapping) to separate public and private services, so that they would each listen on their own Ambassador port. Then I attached a different LoadBalancer service to each port. At the cloud provider side, an IP-based whitelisting approach was used to assign rules directly to the load balancers, to ensure that public or private traffic was firewalled for each load balancer.

But the downside of this approach is that you're stuck with L4 load balancing, and unable to use Mappings, which unlock the full power and featureset of L7 load balancing :(
